### PR TITLE
[PW-1176 ]Add application info for terminal api 

### PR DESCRIPTION
**Pull request by AlexandrosMor**
Add application info for terminal api 
Add SaleToAcquirerDataModel
Modify nexo SaleData model

### CODE REVIEW

File: Adyen.Test/SaleToAcquirerDataTest.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+using Adyen.Model.Terminal;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections.Generic;
+
+namespace Adyen.Test
+{
+    [TestClass]
+    public class SaleToAcquirerDataTest
+    {
+        //private readonly string json = "{\"metadata\":{\"key\":\"value\"},\"shopperEmail\":\"myemail@mail.com\",\"shopperReference\":\"13164308\",\"recurringContract\":\"RECURRING,ONECLICK\",\"shopperStatement\":\"YOUR SHOPPER STATEMENT\",\"recurringDetailName\":\"VALUE\",\"recurringTokenService\":\"VALUE\",\"store\":\"store value\",\"merchantAccount\":\"merchantAccount\",\"currency\":\"EUR\",\"applicationInfo\":{\"adyenLibrary\":{\"name\":\"" + Adyen.Constants.ClientConfig.LibName + "\",\"version\":\"" + Adyen.Constants.ClientConfig.LibVersion + "\"},\"externalPlatform\":{\"integrator\":\"externalPlatformIntegrator\",\"name\":\"externalPlatformName\",\"version\":\"2.0.0\"},\"merchantDevice\":{\"os\":\"merchantDeviceOS\",\"osVersion\":\"10.12.6\",\"reference\":\"4c32759faaa7\"}},\"tenderOption\":\"ReceiptHandler,AllowPartialAuthorisation,AskGratuity\",\"additionalData\":{\"key.key\":\"value\",\"key.keyTwo\":\"value2\"}}";
```
Comment (Aleffio) [inline Adyen.Test/SaleToAcquirerDataTest.cs]:
```suggestion
      
```

File: Adyen/Model/Terminal/SaleToAcquirerData.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            sb.Append("  ShopperReference: ").Append(ShopperReference).Append("\n");
+            sb.Append("  RecurringContract: ").Append(RecurringContract).Append("\n");
+            sb.Append("  ShopperStatement: ").Append(ShopperStatement).Append("\n");
+            sb.Append("  RecurringDetailName: ").Append(RecurringDetailName).Append("\n");
+            sb.Append("  RecurringTokenService: ").Append(RecurringTokenService).Append("\n");
+            sb.Append("  Store: ").Append(Store).Append("\n");
+            sb.Append("  MerchantAccount: ").Append(MerchantAccount).Append("\n");
+            sb.Append("  Currency: ").Append(Currency).Append("\n");
+            sb.Append("  ApplicationInfo: ").Append(ApplicationInfo).Append("\n");
+            sb.Append("  TenderOption: ").Append(Currency).Append("\n");
```
Comment (Aleffio) [inline Adyen/Model/Terminal/SaleToAcquirerData.cs]:
```suggestion
            sb.Append("  TenderOption: ").Append(TenderOption).Append("\n");
```

File: Adyen/Model/Terminal/SaleToAcquirerData.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        }
+
+        /// <summary>
+        /// Returns the string presentation of the object
+        /// </summary>
+        /// <returns>String presentation of the object</returns>
+        public override string ToString()
+        {
+            var sb = new StringBuilder();
+            sb.Append("class Company {\n");
```
Comment (Aleffio) [inline Adyen/Model/Terminal/SaleToAcquirerData.cs]:
```suggestion
            sb.Append("class SaleToAcquirerData {\n");
```

**Comment by rikterbeek** [comment]
@AlexandrosMor missing: public SaleData() {
        if (saleToAcquirerData == null) {
            saleToAcquirerData = new SaleToAcquirerData();
        }
    }

**Pull request merged**